src/Behat/Context/FileContext.php

```diff
@@ -46,4 +46,40 @@ public function createFileWithName($filename, $directory = NULL) {
     $this->files[] = $this->getCore()->createFileWithName($path, $directory);
   }
 
-}
\ No newline at end of file
+  /**
+   * Get path for file in drupal 8.
+   *
+   * @param string $filename
+   *   The name of the file to get.
+   * @param string directory
+   *   A string containing the files scheme, usually "public://".
+   *
+   * @throws Exception
+   *   Exception file not found.
+   *
+   * @throws Exception
+   *   Exception file could not be copied.
+   *
+   * @Given file get name :filename
+   * @Given file get name :filename in the :directory directory
+   */
+  public function visitFileWithName($filename, $directory = NULL) {
+    $public = 'public://';
+    $private = 'private://';
+
+    if (empty($directory) || $directory === $public) {
+      $realpath = \Drupal::service('file_system')->realpath($directory);
+      $path = str_replace(DRUPAL_ROOT, '', $realpath);
+      $destination = $path . '/' . basename($filename);
+    }
+
+    if (!empty($directory) && $directory === $private) {
+      $destination = \Drupal\Core\Url::fromRoute('system.private_file_download', ['filepath' => $filename], [
+        'relative' => TRUE,
+      ])->toString();
+    }
+
```

Comment:
This should throw an exception when $destination is not defined, or it will fire an exception.

Reply:
Thanks